app/src/org/commcare/activities/LoginActivity.java

```diff
@@ -154,15 +154,23 @@ protected void onSaveInstanceState(Bundle savedInstanceState) {
      *                       upon successful login
      */
     protected void initiateLoginAttempt(boolean restoreSession) {
-        if (uiController.getEnteredPasswordOrPin().equals("")) {
-            raiseLoginMessage(StockMessages.Auth_EmptyPassword, false);
-            return;
+        LoginMode loginMode = uiController.getLoginMode();
+
+        if ("".equals(uiController.getEnteredPasswordOrPin()) &&
+                loginMode != LoginMode.PRIMED) {
+            if (loginMode == LoginMode.PASSWORD) {
+                raiseLoginMessage(StockMessages.Auth_EmptyPassword, false);
+                return;
+            } else {
+                raiseLoginMessage(StockMessages.Auth_EmptyPin, false);
+                return;

```

Comment:
Can you lift the `return` from both branches of the if/else out below the conditional? It eliminates the need for 2 `return` statements.